llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -1159,6 +1159,28 @@ getRangeViaSLT(CmpInst::Predicate Pred, APInt RHS,
   return std::nullopt;
 }
 
+/// Get value range for a "ctpop(Val) Pred RHS" condition.
+static ValueLatticeElement getValueFromICmpCtpop(ICmpInst::Predicate Pred,
+                                                 Value *RHS) {
+  unsigned BitWidth = RHS->getType()->getPrimitiveSizeInBits();
```

Comment:
```suggestion
  unsigned BitWidth = RHS->getType()->getScalarSizeInBits();
```

Reply:
Resloved, but may I know the reason to use `getScalarSizeInBits` instead of `getPrimitiveSizeInBits`?

Reply:
getScalarSizeInBits() also works on vectors. It's not relevant here (unless maybe for select conditions?) but most code in LLVM handles scalars and vectors, so it's best to always use this API.